io/net/irqbalance.py

```diff
@@ -0,0 +1,377 @@
+        '''
+        Selects single IRQ number of device and sets,
+        a. Assign all the avialable CPU's serially from min to maximum
+           avaliable CPU's and validates the operations.
+        Eg: 1
+            1,2
+            1,2, ----> 99 [ upto max available CPU's ]
+
+        b. Unassing all the avialable CPU's serially from max to minimum
+           avialble CPU's and validates the operations.
+        Eg: 1,2 -----> 99
+            1,2 ----> 98
+            1 [ upto min avialable CPU's ]
+        '''
+        self.get_device_interrupts()
+        self.get_irq_numbers()
+
+        if len(self.irq_number) == 1:
+            self.irq_number = self.irq_number[0]
+        else:
+            self.irq_number = self.irq_number[1]
+
+        # Adding CPU's to IRQ's serailly upto max available CPU's
+        for cpu_number in range(len(self.cpu_list)):
+            self.cpu_range = self.cpu_list[:cpu_number+1]
+            process.run(f'echo {str(self.cpu_range)[1:-1]} > /proc/irq/'
+                        f'{self.irq_number}/smp_affinity_list',
+                        shell=True, ignore_status=True)
+            wait.wait_for(self.grep_module_interrupts, timeout=5)
+            if len(self.cpu_range) == 1 or len(self.cpu_range) > 1:
+                self.cpu_range_validation()
+        self.collect_dmesg_logs()
+
+        # un-assgining CPU's in reverse order upto minimum available CPU's
+        while len(self.cpu_range) > 1:
+            cpu_range = ','.join(str(n) for n in self.cpu_range)
+            process.run(f'echo {str(cpu_range)} > /proc/irq/'
+                        f'{self.irq_number}/smp_affinity_list',
+                        shell=True, ignore_status=True)
+            if len(self.cpu_range) > 1:
+                self.cpu_range_validation()
+            wait.wait_for(self.grep_module_interrupts, timeout=5)
+            self.cpu_range = self.cpu_range[:-1]
+        self.collect_dmesg_logs()
+
+    def cpu_range_validation(self):
+        '''
+        Funtion to validate the assinged CPU's by script.
+        '''
+        self.irq_affinity = '-'.join([str(self.cpu_range[0]),
+                                      str(self.cpu_range[-1])])
+        self.system_affinity = process.system_output(
+            f'cat /proc/irq/{self.irq_number}/smp_affinity_list',
+            shell=True,
+            ignore_status=True
+            ).decode("utf-8")
+        if not (self.irq_affinity == self.system_affinity or
+                len(self.cpu_range) == len(self.system_affinity)):
+            self.fail(f'The smp_affinity_list does not matches to affinity'
+                      ' that was set by script, Please check the logs')
+
+    def grep_module_interrupts(self):
+        '''
+        Funtion to filter all interrupts along associated CPU's of device.
+        '''
+        process.run(
+            f'head -n 1 /proc/interrupts &&'
+            f' grep {self.interface_type} /proc/interrupts',
```

Comment:
it will look clean if it has cmd and process in 2 different line.. also a common rules is to return from a method